Feature/add cat points

frozenhelium opened the pull request:
Depends on https://github.com/toggle-corp/react-store/pull/615

frozenhelium committed:
Add risk points and info on hover

frozenhelium committed:
Add hover details for the risk points for districts

frozenhelium committed:
Add hover details for palika risk points

frozenhelium committed:
Add labels to the regions

frozenhelium committed:
Add line string between risk point and relocation points

thenav56 committed:
Fix triagram search bug

thenav56 committed:
Add caching for metadata APIs

tnagorra commented on react/src/Root/App/DistrictOverview/index.tsx:
```diff
@@ -107,14 +112,150 @@ class DistrictOverview extends React.PureComponent<MyProps, State> {
         });
     }
 
+    private getLabelGeoJson = memoize((metadata?: Metadata) => {
```
Instead of `private getLabelGeoJson = memoize((metadata?: Metadata)`, we should use this form:
`private getLabelGeoJson = memoize((metadata: Metadata | undefined)`

tnagorra commented on react/src/Root/App/DistrictOverview/index.tsx:
```diff
@@ -107,14 +112,150 @@ class DistrictOverview extends React.PureComponent<MyProps, State> {
     }
 
+    private getLabelGeoJson = memoize((metadata?: Metadata) => {
+        const { regions = [] } = metadata || {};
```
We need to fix this.

tnagorra commented on react/src/Root/App/DistrictOverview/index.tsx:
```diff
@@ -154,8 +296,35 @@ class DistrictOverview extends React.PureComponent<MyProps, State> {
             ];
         }
 
+        const districtMetadata = (response as Metadata) || {};
```
We should not do this.

tnagorra commented on react/src/Root/App/DistrictOverview/index.tsx:
```diff
@@ -154,8 +296,35 @@ class DistrictOverview extends React.PureComponent<MyProps, State> {
+        const cat2PointsGeoJson = this.convertRiskPointToGeoJson(cat2Points);
+        const cat3PointsGeoJson = this.convertRiskPointToGeoJson(cat3Points);
+        const {
+            pla: plaRelocationPointGeoJson,
```
Are these names `pla` and `is` sent from the server?

tnagorra commented on react/src/Root/App/NationalOverview/index.tsx:
```diff
@@ -79,6 +80,31 @@ function wrapInArray<T>(item?: T) {
+    const geojson = {
+        type: 'FeatureCollection',
+        features: geoAttributeList
+            .map(level => ({
```
Instead of having default centroid, we should filter out points without centroid.

tnagorra commented on react/src/Root/App/NationalOverview/index.tsx:
```diff
@@ -206,6 +232,48 @@ class NationalOverview extends React.PureComponent<MyProps, State> {
     }
 
+    private getLabelGeoJson = memoize((metadata?: Metadata) => {
+        const { regions = [] } = metadata || {};
```
We shouldn't do this.

tnagorra commented on react/src/Root/App/PalikaOverview/index.tsx:
```diff
@@ -17,115 +13,223 @@ import {
-     */
+
+    private getLabelGeoJson = memoize((metadata?: Metadata) => {
+        const { regions = [] } = metadata || {};
```
We should not do this.

tnagorra commented on react/src/Root/App/PalikaOverview/index.tsx:
```diff
@@ -140,38 +244,56 @@ class DistrictOverview extends React.PureComponent<MyProps, State> {
+            ];
+        }
+
+        const palikaMetadata = response as Metadata || {};
```
We shouldn't do this

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +41,162 @@ export const sanitizeResponse = (data: unknown): any => {
+                    ],
+                },
+                properties: {
+                    ...catPoint,
```
We should only add property which is necessary.

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +41,162 @@ export const sanitizeResponse = (data: unknown): any => {
+    relocationPointList: RelocationPoint[] | undefined = [],
+) {
+    const plaRelocationPointList = relocationPointList.filter(
+        d => d.solutionType === 'Private Land Acquisition',
```
This looks very fragile

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +41,162 @@ export const sanitizeResponse = (data: unknown): any => {
+    );
+
+    const isRelocationPointList = relocationPointList.filter(
+        d => d.solutionType === 'Integrated Settlement',
```
This looks very fragile

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +41,162 @@ export const sanitizeResponse = (data: unknown): any => {
+                    coordinates: relocationPoint.location,
+                },
+                properties: {
+                    ...relocationPoint,
```
We should only add property which is necessary.

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +41,162 @@ export const sanitizeResponse = (data: unknown): any => {
+                    coordinates: relocationPoint.location,
+                },
+                properties: {
+                    ...relocationPoint,
```
We should only add property which is necessary.

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +41,162 @@ export const sanitizeResponse = (data: unknown): any => {
+                geometry: {
+                    type: 'Point',
+                    coordinates: [
+                        ...(level.centroid || []),
```
We should filter this out instead

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +41,162 @@ export const sanitizeResponse = (data: unknown): any => {
+    const filteredRelocationPointList = relocationPointList
+        .filter(d => cat2Points[d.geosite] || cat3Points[d.geosite])
+        .filter(d => (
+            d.solutionType === 'Private Land Acquisition'
```
This looks very fragile

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +41,162 @@ export const sanitizeResponse = (data: unknown): any => {
+            .map((relocationPoint, i) => {
+                const coordinates = [];
+
+                coordinates.push([...relocationPoint.location || []]);
```
We shouldn't push location that is undefined.

tnagorra reviewed (changes requested):
Please fix problems with default values

frozenhelium committed:
WIP

frozenhelium committed:
Add hover sync between points

tnagorra commented on react/src/Root/App/DistrictOverview/index.tsx:
```diff
@@ -49,21 +72,102 @@ const requests: { [key: string]: ClientAttributes<Props, Params> } = {
+    featureFromIdentifier: FeatureFromIdentifier,
+    lineStringIdentifier,
+) => {
+    const newMapState = [] as MapStateElement[];
```
Should be written as
`const newMapState: MapStateElement[] = [];`

tnagorra commented on react/src/Root/App/DistrictOverview/index.tsx:
```diff
@@ -49,21 +72,102 @@ const requests: { [key: string]: ClientAttributes<Props, Params> } = {
+
+    catPointList.forEach((catPoint) => {
+        const shouldDim = id ? catPoint.geosite !== geosite : false;
+        const rs = catPoint.relocationSites;
```
This should be destructured

tnagorra commented on react/src/Root/App/DistrictOverview/index.tsx:
```diff
@@ -49,21 +72,102 @@ const requests: { [key: string]: ClientAttributes<Props, Params> } = {
+        const shouldDim = id ? catPoint.geosite !== geosite : false;
+        const rs = catPoint.relocationSites;
+
+        let relocationSites = [] as RelocationSite[];
```
Should be written as
`const relocationSites: RelocationSite[] = [];`

tnagorra commented on react/src/Root/App/DistrictOverview/index.tsx:
```diff
@@ -49,21 +72,102 @@ const requests: { [key: string]: ClientAttributes<Props, Params> } = {
+        };
     }
 
+    private featureIdentifier = {}
```
Shouldn't there be types for `featureIdentifier`, `featureFromIdentifier` and `lineStringIdentifier`

frozenhelium committed:
Change font

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +45,322 @@ export const sanitizeResponse = (data: unknown): any => {
+    cat2PointList: RiskPoint[] | undefined = [],
+    cat3PointList: RiskPoint[] | undefined = [],
+) {
+    const catPointList = [
```
Sholdn't cast unnecessarily

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +45,322 @@ export const sanitizeResponse = (data: unknown): any => {
+
+    const featureIdentifier: FeatureIdentifiers = {};
+
+    let count = 1;
```
Having a separate count can be avoided

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +45,322 @@ export const sanitizeResponse = (data: unknown): any => {
+                const validRelocationSites = catPoint.relocationSites
+                    .filter(d => !!d.latitude && !!d.longitude);
+
+                if (validRelocationSites.length === 0) {
```
We should use filter instead of returning null

tnagorra commented on react/src/utils/common.ts:
```diff
@@ -33,3 +45,322 @@ export const sanitizeResponse = (data: unknown): any => {
+                    ))
+                );
+            },
+        ).flat().filter(d => d),
```
We can skip this filter

frozenhelium committed:
Add more details to the information pane

* Remove `LandslidesRiskScore`
* Add `IntegratedSettlements`
* Add `EarthquakeAffectedLandlessHouseholds`
* Add `TrancheUpdate`

tnagorra committed:
Remove un-necessary casting

tnagorra committed:
Add polyfill in build pipeline

- Update all dependencies

frozenhelium committed:
Pull tranche, integrated settlement and landless data

tnagorra committed:
Change font in map

tnagorra committed:
Move components and resuable functions

tnagorra committed:
Refactor DistrictOverview

- Disable selection and hover on map while data loading
- Enable caching

tnagorra committed:
Remove mapState from state

- Clear out hover ids while setting selection id (hover out wasn't
  called after selection)

tnagorra committed:
Change color for connection lines

tnagorra committed:
Refactor logic for geojson generation and mapState generation

- Add logic for mapState when relocation points selected
- Remove unused comments
- Switch to different paint when hovered/selected
- Move position of controls in map

tnagorra committed:
Change centroid for georegions

frozenhelium committed:
Add RelocationPointDetail component

tnagorra committed:
Create handler creator for selection and hover

tnagorra committed:
Use same component for District and Ward

frozenhelium committed:
Fix cat point details

frozenhelium committed:
Add numberOfHouseholds in relocation site response

tnagorra committed:
Show detail on hover/selection for related points

frozenhelium committed:
Add layer switcher button

thenav56 committed:
Add deploy config

tnagorra committed:
Add map layers
